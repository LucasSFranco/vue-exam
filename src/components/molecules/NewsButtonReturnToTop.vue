<template>
  <button
    class="button return-to-top"
    @click="returnToTop"
  >
    <Icon icon="angle-double-up" />
  </button>
</template>

<script>

export default {
  name: 'NewsButtonReturnToTop',
  methods: {
    returnToTop() {
      document.querySelector('.scrollbox').scrollTop = 0
    },
  },
}

</script>

<style lang="scss" scoped>

  .button.return-to-top {
    width: 2.5rem;
    height: 2.5rem;

    background: var(--button-normal);
    color: var(--button-color);
    cursor: pointer;
    font-size: 1rem;

    display: flex;
    align-items: center;
    justify-content: center;

    transition: 100ms background-color;

    position: absolute;
    left: 1.5rem;
    bottom: 1.5rem;

    &:hover {
      background: var(--button-hover);
    }

    &:active {
      background: var(--button-active);
    }
  }

</style>
