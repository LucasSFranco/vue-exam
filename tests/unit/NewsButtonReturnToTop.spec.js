import { shallowMount } from '@vue/test-utils'

import NewsButtonReturnToTop from '@/components/molecules/NewsButtonReturnToTop.vue'

describe('News Button Return To Top', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(NewsButtonReturnToTop, {
      global: {
        stubs: {
          Icon: true
        }
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
