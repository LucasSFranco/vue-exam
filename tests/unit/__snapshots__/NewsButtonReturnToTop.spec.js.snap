// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News Button Return To Top matches snapshot 1`] = `
<button class="button return-to-top">
  <icon-stub icon="angle-double-up"></icon-stub>
</button>
`;
